Data/Euclidean.hs

```diff
@@ -142,6 +143,8 @@ class GcdDomain a => Euclidean a where
   -- prop> \x y -> y == 0 || let (q, r) = x `quotRem` y in (r == 0 || degree r < degree y)
   degree :: a -> Natural
 
+  {-# MINIMAL (quotRem | quot, rem), degree #-}
```

Comment:
this MINIMAL pragma will need some CPP, see elsewhere in the project for examples